### ISSUE TITLE
Add Rating Bar in Course Cards

### ISSUE THREAD

**Issue by aggarwalpulkit596** [coding-blocks/CBOnlineApp#219]

**Comment by shubhamofbce** [comment]
@aggarwalpulkit596 
I would like to work on this issue

**Comment by adityastic** [comment]
@shubhamofbce are you working on this?

**Comment by shubhamofbce** [comment]
@adityastic please go ahead i tried a lot on this issue but was unable to solve this and also exams came at the same time..

**Comment by adityastic** [comment]
Thanks

**Comment by adityastic** [comment]
@aggarwalpulkit596 so the rating bar should be available for everyone? Or for only the people who are subscribed to that course and also can you link me the api for rating the course ?

**Comment by aggarwalpulkit596** [comment]
You have to add a rating bar in anko course card which shows the current rating of course and the feature you talking about is also not there you can open a issue and add that as well

**Comment by adityastic** [comment]
@aggarwalpulkit596 let me just add a rating bar for now. I shall open an issue on the server and then integrate into the app as well the next week

**Comment by aggarwalpulkit596** [comment]
It exists on server already it's haven't been integrated in the app

**Comment by adityastic** [comment]
@aggarwalpulkit596 oh great then, I will open an issue to add it to the app then later this week

**Comment by HardikSo** [comment]
is this issue still open?

**Comment by adityastic** [comment]
@adityastic I haven't worked on it yet, But I plan to later today :)

**Comment by adityastic** [comment]
@aggarwalpulkit596 currently the Course Cards are an AnkoComponent. Can I change it to the legacy xml like the one used in MyCourseCard like in MyCourses ?

**Comment by aggarwalpulkit596** [comment]
Obviously not we would be moving to anko layouts completely in future

**Comment by adityastic** [comment]
@aggarwalpulkit596 well the theming option for ratingbar does not work. How can I make them small? changing height or width does not work.

**Comment by adityastic** [comment]
@aggarwalpulkit596 Update: Anko is Amazing, But it has some issues as well. I spent last night on this and was able to get it working.

**Comment by aggarwalpulkit596** [comment]
I Guess themedratingbar was the solution ?

**Comment by adityastic** [comment]
@aggarwalpulkit596 no it wasn't, I had to create a small layout file and include in as a ankocomponent. Unfortunately there were no resources on this issue on the internet, After doing some research for hours this was the only working solution

**Comment by adityastic** [comment]
I tried themedRatingBar, creating customviews and creating inline functions before asking you about shifting to legacy xml. But they don't take in the style attribute as it does not come under the attribute set parameter

**Comment by shubhamofbce** [comment]
Yes i also faced the same issues and then handedover this to you @adityastic .

**Comment by adityastic** [comment]
@shubhamofbce it's done :), now im waiting for @aggarwalpulkit596 to finalize the other prs.

**Comment by shubhamofbce** [comment]
@adityastic what was the solution ???

**Comment by aggarwalpulkit596** [comment]
@adityastic running busy lately will.merge all.of them by today 😅

**Comment by adityastic** [comment]
@shubhamofbce So, as specified in the docs of anko. RatingBar can be used as :
`ratingBar{}`
and as we wanted a small ratingBar, we had to add the style attribute `Widget.AppCompat.Ratingbar.Small` to it. As suggested in the anko docs, we can use there extension function 
`themedRatingBar(R.style.WhateverTheme){}`
But this does work, as Theme Wrapped inside ContextThemeWrapper does not work with RatingBars.

So what I ended up doing was, I created a layout for RatingBar with style attribute and included it as a AnkoComponent like this:
`courseRatingBar = include(R.layout.ratingbar_small)`  
this way it retained the style attribute. 

Here is how the final product looks :
<img src="https://user-images.githubusercontent.com/11988517/58813563-a6be5b00-8641-11e9-912a-456c4433d7a1.jpg" width= "300" />

**Comment by aggarwalpulkit596** [comment]
Did you try this https://github.com/Kotlin/anko/issues/16

**Comment by adityastic** [comment]
@aggarwalpulkit596 this feature is already implemented in the library. This is the reason why themedRatingBar or any other themed component exists

**Comment by adityastic** [comment]
@aggarwalpulkit596 the actual issue can be found here https://github.com/Kotlin/anko/issues/361

**Comment by aggarwalpulkit596** [comment]
Got it

**Comment by adityastic** [comment]
@aggarwalpulkit596 Wow, so I just created an extension function that can solve https://github.com/Kotlin/anko/issues/361

**Comment by aggarwalpulkit596** [comment]
Cool make a PR then

**Comment by aggarwalpulkit596** [comment]
Because if you wouldn't have done it using extension then i probably wouldn't have accepted this PR 😅 as i want to get rid of xml layouts

**Comment by adityastic** [comment]
@aggarwalpulkit596 haha yes, Possiblities with anko are inevitable. Please create issues with shifting everything to anko. This is amazing, and I would to spend my time researching on this library. Btw I dropped a PR for  fixing the build and the rating bar

**Comment by aggarwalpulkit596** [comment]
Not right now waiting for mew designs then I'll think about it and used a 4 parameters constructor right there was an answer similar to that

**Comment by adityastic** [comment]
@aggarwalpulkit596 yes, certain components in API 21+ shifted to a 4 parameter constructor for the StyleRes. This Is where I got to know about it https://developer.android.com/reference/android/view/View.html#View(android.content.Context,%2520android.util.AttributeSet,%2520int,%2520int). Then I used the solution in https://github.com/Kotlin/anko/issues/724 to create a extension function, easy xD
